common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    @Override
+    public void setClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> setClaimMarker(claim, world));
+    }
+
+    @Override
+    public void removeClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> removeClaimMarker(claim, world));
+    }
+
+    @Override
+    public void clearAllMarkers() {
+        claims.clear();
+    }
+
+    @NotNull
+    private String getLayerName() {
+        return plugin.getSettings().getWebMapMarkerSetName();
+    }
+
+    @NotNull
+    private String getClaimMarkerKey(@NotNull TownClaim claim, @NotNull net.pl3x.map.core.world.World world) {
+        return plugin.getKey(
+            claim.town().getName().toLowerCase(),
+            Integer.toString(claim.claim().getChunk().getX()),
+            Integer.toString(claim.claim().getChunk().getZ()),
+            world.getName()
+        ).toString();
+    }
+
+    private void registerLayers(@NotNull net.pl3x.map.core.world.World world) {
+        if (plugin.getSettings().doWebMapHook()) {
+            ClaimsLayer layer = new ClaimsLayer(this, world);
+            world.getLayerRegistry().register(layer);
+        }
+    }
+
+    @EventHandler
+    public void onPl3xMapEnabled(@NotNull Pl3xMapEnabledEvent event) {
+        // Register layers for each world
+        Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
+    }
+
+    @EventHandler
+    public void onWorldLoaded(@NotNull WorldLoadedEvent event) {
+        registerLayers(event.getWorld());
+    }
+
+    @EventHandler
+    public void onWorldUnloaded(@NotNull WorldUnloadedEvent event) {
+        event.getWorld().getLayerRegistry().unregister(CLAIMS_LAYER);
+    }
+
+    @NotNull
+    public Options getMarkerOptions(@NotNull TownClaim claim) {
+        return Options.builder()
+            .tooltip(new Tooltip(claim.town().getName()).setDirection(Tooltip.Direction.TOP))
+            // TODO: Unsure what color settings are usually used so left it very bare-bones
+            .fillColor(claim.town().getColor().getRGB())
+            .strokeColor(claim.town().getColor().darker().getRGB())
+            .strokeWeight(1)
+            .build();
+    }
+
+    public static class ClaimsLayer extends SimpleLayer {
+
+        private final Pl3xMapHook hook;
+        private final net.pl3x.map.core.world.World mapWorld;
+
+        public ClaimsLayer(@NotNull Pl3xMapHook hook, @NotNull net.pl3x.map.core.world.World mapWorld) {
+            super(CLAIMS_LAYER, hook::getLayerName);
+            this.hook = hook;
+            this.mapWorld = mapWorld;
+        }
+
+        @Override
+        @NotNull
+        public Collection<Marker<?>> getMarkers() {
+            return hook.claims.stream()
+                // TODO: Need a way to filter claim chunks by world
+//                .filter(claim -> claim.claim().getChunk().getWorld().getName().equals(mapWorld.getName()))
```

Comment:
see above -- will need to make that queue a map and then iterate through the entries by world key :)